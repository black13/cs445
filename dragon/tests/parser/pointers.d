program example();
var x, y: integer;
var a, b, c: ^integer;
begin
    a^ := x;
    b := a;
    c := @y
end.
