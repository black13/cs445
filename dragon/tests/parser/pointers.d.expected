PROGRAM `example` WITH ARGS:
AND VARIABLES:
  DECLARATION WITH TYPE:
    INTEGER
  OF NAMES:
    `x`
    `y`
  DECLARATION WITH TYPE:
    POINTER TO:
      INTEGER
  OF NAMES:
    `a`
    `b`
    `c`
AND SUBPROGRAMS:
DOES:
  ASSIGN
    DEREF:
      a
  TO
    x
  ASSIGN
    b
  TO
    a
  ASSIGN
    c
  TO
    ADDRESS OF:
      y
-- done dumping ast --
